### PR TITLE
Feat rspy153/implicit collection owner

### PR DESCRIPTION
**Pull request by agrosu1978**
Implementation of the optional ownerId for all catalog endpoints

### CODE REVIEW

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 1]
```diff
@@ -604,10 +604,8 @@ async def manage_get_response(  # pylint: disable=too-many-locals, too-many-bran
         ):
             detail = {"error": "Unauthorized access."}
             return JSONResponse(content=detail, status_code=HTTP_401_UNAUTHORIZED)
-        elif "/queryables" in request.scope["path"]:
-            content["$id"] = request.url._url  # pylint: disable=protected-access
         elif (
-            "/collections" in request.scope["path"] and "items" not in request.scope["path"]
+            "/collection" in request.scope["path"] and "items" not in request.scope["path"]
```
Comment (Padeanu) [inline services/catalog/rs_server_catalog/user_catalog.py:608]: Typo check:
previously it was collection**s**

File: services/catalog/tests/test_authentication_catalog.py
Comment (Padeanu) [inline services/catalog/tests/test_authentication_catalog.py:1]: Missing test case where 2 collections from 2 different users have the same name (Toto:S1_L1, pyteam:S1_L1) and test that GET /collections with apikey actually selects the correct collections.

Comment (Padeanu) [reply services/catalog/tests/test_authentication_catalog.py:1]: Also permission should allow both users to read both collection in order to make test meaningfull.

**Comment by Padeanu** [comment]
https://github.com/RS-PYTHON/rs-server/blob/develop/docs/doc/users/catalog.md should be updated aswell

**Review by jgaucher-cs — CHANGES_REQUESTED** [review]
Should the `owner_id` become optional in all the catalog endpoints ?

image

**Review by cyouri-csgroup — APPROVED** [review]

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 1 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py]: Solved

**Pull request merged**